src/master-playlist-controller.js

```diff
@@ -1064,26 +1064,37 @@ export class MasterPlaylistController extends videojs.EventTarget {
     }
 
     let isFinalRendition = this.masterPlaylistLoader_.isFinalRendition_();
+    let playlists = this.masterPlaylistLoader_.master.playlists;
 
-    if (isFinalRendition) {
-      // Never blacklisting this playlist because it's final rendition
+    if (playlists.length === 1) {
+      // Never blacklisting this playlist because it's the only playlist
       videojs.log.warn('Problem encountered with the current ' +
-                       'HLS playlist. Trying again since it is the final playlist.');
+                       'HLS playlist. Trying again since it is the only playlist.');
 
       this.tech_.trigger('retryplaylist');
       return this.masterPlaylistLoader_.load(isFinalRendition);
     }
+    if (isFinalRendition) {
+      // Since we're on the final non-blacklisted playlist, and we're about to blacklist it,
+      // instead of erring the player or retrying this playlist, we clear out the current blacklist
+      // so the other playlists may be attempted in case any have been fixed.
+      playlists.forEach((playlist) => {
+        delete playlist.excludeUntil;
+      });
+      videojs.log.warn('Clearing blacklist for every playlist because last rendition ' +
```

Comment:
Minor, but putting it just before the loop might be better for a couple reasons: (1) if something breaks (throws an exception) during the loop, we know the last warning log and it can help us determine what broke and (2) we already cleared everything, so `Clearing` might be more accurate before